src/test/java/emissary/core/channels/LoggingInputStreamFactoryTest.java

```diff
@@ -0,0 +1,127 @@
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class LoggingInputStreamFactoryTest {
+    private ListAppender<ILoggingEvent> appender = null;
+    private final Logger logger = (Logger) LoggerFactory.getLogger(LoggingInputStreamFactoryTest.class);
+
+    @BeforeEach
+    void setup() throws IOException {
+        appender = new ListAppender<>();
+        appender.start();
+        logger.addAppender(appender);
+    }
+
+    @AfterEach
+    void detach() throws Exception {
+        logger.detachAppender(appender);
+    }
+
+    @Test
+    void testChannelFactory() throws IOException {
+        final byte[] bytes = new byte[100];
+        final TestInputStreamFactory tisf = new TestInputStreamFactory(bytes);
+        final InputStreamFactory loggingFalseIsf = LoggingInputStreamFactory.create(tisf, "Identifier", logger, false);
+
+        testLogging(appender, loggingFalseIsf, false);
+
+        final InputStreamFactory loggingTrueIsf = LoggingInputStreamFactory.create(tisf, "Identifier", logger, true);
+
+        testLogging(appender, loggingTrueIsf, true);
+    }
+
+    private static void testLogging(final ListAppender<ILoggingEvent> appender, final InputStreamFactory isf,
+            final boolean testStackTrace) throws IOException {
+        assertTrue(appender.list.isEmpty());
+
+        try (InputStream is = isf.create()) {
+            check(appender, "Identifier : 0 : created : lST=" + testStackTrace, testStackTrace);
+
+            is.markSupported();
+
+            check(appender, "Identifier : 0 : markSupported : r=true", testStackTrace);
+
+            is.mark(100);
+
+            check(appender, "Identifier : 0 : mark : rl=100", testStackTrace);
+
+            is.read();
+
+            check(appender, "Identifier : 0 : read-0 : r=0", testStackTrace);
+
+            is.read(new byte[10]);
+
+            check(appender, "Identifier : 0 : read-1 : r=10", testStackTrace);
+
+            is.read(new byte[10], 5, 5);
+
+            check(appender, "Identifier : 0 : read-3 : off=5 len=5 : r=5", testStackTrace);
+
+            is.available();
+
+            check(appender, "Identifier : 0 : available : r=84", testStackTrace);
+
+            is.skip(15);
+
+            check(appender, "Identifier : 0 : skip : n=15 : r=15", testStackTrace);
+
+            is.reset();
+
+            check(appender, "Identifier : 0 : reset", testStackTrace);
+        }
+
+        check(appender, "Identifier : 0 : close", testStackTrace);
+    }
+
+    private static void check(final ListAppender<ILoggingEvent> appender, final String message,
```

Comment:
Suggest removing `static` modifier and `appender` parameter.

Reply:
Changed.